src/lib.rs

```diff
@@ -1,59 +1,94 @@
-extern crate errno;
+#![deny(warnings)]
+#![deny(missing_docs)]
+
+//! # rust-zonename
+//!
+//! The `rust-zonename` crate provides a rust abstraction over the illumos `getzoneid`,
+//! `getzoneidbyname`, and `getzonenamebyid` C library functions.
+
 extern crate libc;
 
+use libc::{c_char, size_t};
 use std::ffi::CString;
-use libc::{c_char,size_t};
-use errno::{Errno,errno};
+use std::io;
 
 mod ffi;
 
+/// Max length of a zone name.
 pub const ZONENAME_MAX: usize = 64;
 
-pub fn getzoneid() -> Result<i32, Errno> {
-    let zoneid = unsafe {
-        ffi::getzoneid()
-    };
+/// Returns the zone ID of the calling process.
+///
+/// # Example
+/// ```
+/// let zoneid = zonename::getzoneid().expect("failed to get zoneid");
```

Comment:
this is so cool.  i didn't realize the documentation would also be used for testing that's insanely awesome.

Reply:
Yeah, these are called "doc tests".  Useful for making sure the examples you provide actually do what you think they do.